[CPDLP-2317] Update NPQ outcomes table in the finance application to show details of an successful/unsuccessful request to the TRA

leandroalemao opened the pull request:
### Context

- Ticket: [CPDLP-2317](https://dfedigital.atlassian.net/browse/CPDLP-2317)

### Changes proposed in this pull request

Update NPQ outcomes table in the finance application view to show details of an successful/unsuccessful request to the TRA.


[CPDLP-2317]: https://dfedigital.atlassian.net/browse/CPDLP-2317?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ethax-ross commented on app/components/finance/npq/participant_outcomes/table.rb:
```diff
@@ -16,16 +14,62 @@ def render?
+        end
+
+        def sent_to_tra_at(outcome)
+          return "<strong class='govuk-tag govuk-tag--yellow'>#{t('finance.npq.participant_outcomes.na')}</strong>".html_safe if outcome.sent_to_qualified_teachers_api_at.blank?
```
Worth using `tag.strong()` here and below to avoid the use of `html_safe`?

ethax-ross commented on spec/models/participant_outcome/npq_spec.rb:
```diff
@@ -231,4 +231,135 @@
+      expect(outcome.passed_and_not_sent?).to eql(false)
+    end
+
+    it "returns false if failed" do
```
Would it be worth setting not sent explicitly in these scenarios to make sure its only testing the failed status?

```
outcome = described_class.new(state: "failed", sent_to_qualified_teachers_api_at: nil)
```

ethax-ross commented on spec/models/participant_outcome/npq_spec.rb:
```diff
@@ -231,4 +231,135 @@
+      expect(outcome.passed_and_recorded?).to eql(true)
+    end
+
+    it "returns false if passed and not recorded" do
```
Should we test the other combinations like failed and recorded?

ethax-ross commented on app/components/finance/npq/participant_outcomes/table.rb:
```diff
@@ -16,16 +14,62 @@ def render?
-        def sent_to_tra(outcome)
-          outcome.sent_to_qualified_teachers_api_at&.to_fs(:govuk)
+        def caption_text
+          titles = [t("finance.npq.participant_outcomes.declaration_outcomes")]
```
The use of an array here initially made me think there could be multiple titles... could just be me, but it may be clearer without it 🤷‍♂️ 

```
def caption_text
  title = t("finance.npq.participant_outcomes.declaration_outcomes")
  latest_outcome = outcomes.first

  return title unless latest_outcome.present?
  
  if latest_outcome.passed_and_not_sent?
    outcome_description = t("finance.npq.participant_outcomes.passed")
  elsif latest_outcome.failed_and_not_sent?
    outcome_description = t("finance.npq.participant_outcomes.failed")
  elsif latest_outcome.passed_and_recorded?
    outcome_description = t("finance.npq.participant_outcomes.passed_and_recorded")
  elsif latest_outcome.failed_and_recorded?
    outcome_description = t("finance.npq.participant_outcomes.failed_and_recorded")
  elsif latest_outcome.passed_and_not_recorded?
    outcome_description = t("finance.npq.participant_outcomes.passed_and_not_recorded")
  elsif latest_outcome.failed_and_not_recorded?
    outcome_description = t("finance.npq.participant_outcomes.failed_and_not_recorded")
  end

  "#{title}: #{outcome_description}"
end
```

ethax-ross reviewed (commented):
Looks good, just a few minor suggestions. I'm also not sure how to test/get to this in the UI though so I haven't been able to test it as a user.

leandroalemao commented:
Tks for your review @ethax-ross .. I've addressed your comments on https://github.com/DFE-Digital/early-careers-framework/pull/3825/commits/e9f92cfe54874585cb54e3fb8c465bb2bd16eef1 tks

ethax-ross reviewed (approved):
Looks good to me 👍 I'm not sure how to test the front-end though so I haven't checked that

cwrw commented on app/components/finance/npq/participant_outcomes/table.rb:
```diff
@@ -16,16 +14,61 @@ def render?
+        end
+
+        def sent_to_tra_at(outcome)
+          return tag.strong(t("finance.npq.participant_outcomes.na"), class: "govuk-tag govuk-tag--yellow") if outcome.sent_to_qualified_teachers_api_at.blank?
```
I'm not sure about the tag being yellow here, as in the prototype its only on `Recorded by TRA API`, but let's see what PO review says and leave as is for now :)
